### PR TITLE
config init navigation

### PR DESCRIPTION
**Pull request by Vente16**
## Summary

Initial config react-navigation between library screen and detailbook screen

## Screenshots
react-navigator

## Trello Card

https://trello.com/c/WG3iQHQN/13-n-31-introducci%C3%B3n-a-react-navigation

### CODE REVIEW

File: wbooks/src/app/components/Navigator/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Text } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import Library from '@app/screens/Library';
+
+const Stack = createStackNavigator();
+
+const DetailBook = () => <Text>Hello from deatil</Text>;
```
Comment (sfernandez11) [inline wbooks/src/app/components/Navigator/index.tsx:9]: Create a new screen for this => `app/screens/BookDetail`

File: wbooks/src/app/components/Navigator/index.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+function AppNavigator() {
+  return (
+    <NavigationContainer>
+      <Stack.Navigator>
+        <Stack.Screen name="Home" component={Library} options={{ title: 'Library' }} />
+        <Stack.Screen name="Detail" component={DetailBook} />
```
Comment (sfernandez11) [inline wbooks/src/app/components/Navigator/index.tsx:16]: Extract Routes name in a constant file => `constants/routes`:
```
Routes = {
  Home: 'Home',
  Detail: 'Detail'
}
```

File: wbooks/src/app/screens/Library/index.tsx [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
 
 import Book from './components/Book';
 import styles from './styles';
 
-function Library() {
-  const renderBook: ListRenderItem<IBook> = ({ item }) => <Book book={item} />;
+function Library({ navigation }: any) {
   const renderKeyBook = ({ id }: IBook) => `${id}`;
+  const onPressItem = (item: IBook) => {
+    navigation.navigate('Detail', item);
```
Comment (sfernandez11) [inline wbooks/src/app/screens/Library/index.tsx:13]: Use `Route` constant => `navigation.navigate(Routes.Detail, item);`

File: wbooks/src/app/screens/Library/index.tsx [hunk 4]
```diff
@@ -1,21 +1,25 @@
...
+  };
+  const renderBook: ListRenderItem<IBook> = ({ item }) => (
+    <TouchableWithoutFeedback onPress={() => onPressItem(item)}>
+      <Book book={item} />
+    </TouchableWithoutFeedback>
```
Comment (sfernandez11) [inline wbooks/src/app/screens/Library/index.tsx:18]: This is ok but you can pass via props the `onPress` function and then, in the Book component, replace the `View` container with any `TouchableXXX` or `Pressable` component

**Review by sfernandez11 — COMMENTED** [review]
Great work Jhon! Some comments...

File: wbooks/src/app/screens/Library/index.tsx [hunk 5]
```diff
@@ -1,21 +1,25 @@
...
+import { TouchableWithoutFeedback } from 'react-native-gesture-handler';
 import { BOOKS_MOCK } from '@constants/mockBooks';
 import { Book as IBook } from '@interfaces/book';
 
 import Book from './components/Book';
 import styles from './styles';
 
-function Library() {
-  const renderBook: ListRenderItem<IBook> = ({ item }) => <Book book={item} />;
+function Library({ navigation }: any) {
```
Comment (mcavo) [inline wbooks/src/app/screens/Library/index.tsx:10]: Please type props 😄

File: wbooks/src/app/components/Navigator/index.tsx (hunk 2 above)
Comment (sfernandez11) [reply wbooks/src/app/components/Navigator/index.tsx:14]: You forgot this ⚠️ 
```
        <Stack.Screen name={Routes.Home} component={Library} options={{ title: 'Library' }} />
        <Stack.Screen name={Routes.Detail} component={DetailBook} />
```

**Review by sfernandez11 — APPROVED** [review]
🥇

**Pull request merged**